apps/emqx_auth_kerberos/src/emqx_authn_kerberos.erl

```diff
@@ -0,0 +1,121 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+-module(emqx_authn_kerberos).
+
+-include("emqx_auth_kerberos.hrl").
+-include_lib("emqx_auth/include/emqx_authn.hrl").
+-include_lib("typerefl/include/types.hrl").
+
+-behaviour(emqx_authn_provider).
+
+-export([
+    create/2,
+    update/2,
+    destroy/1,
+    authenticate/2
+]).
+
+create(
+    AuthenticatorID,
+    #{
+        principal := Principal,
+        keytab_file := KeyTabFile
+    }
+) ->
+    KeyTabPath = emqx_schema:naive_env_interpolation(KeyTabFile),
+    case sasl_auth:kinit(KeyTabPath, Principal) of
```

Comment:
if we force server to configure default keytab file, there is no need for server to do kinit.

Reply:
we may want to keep it here so errors can be detected sooner if principal is configured wrong for example.